gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -0,0 +1,502 @@
+                failure_reason: format!("Failed get chain tip block summary: {e:?}"),
+            })?
+            .into_iter()
+            .next()
+        else {
+            return Err(LightningRpcError::FailedToGetNodeInfo {
+                failure_reason:
+                    "Failed to get chain tip block summary (empty block list was returned)"
+                        .to_string(),
+            });
+        };
+
+        let esplora_chain_tip_timestamp = chain_tip_block_summary.time.timestamp;
+        let block_height: u32 = chain_tip_block_summary.time.height;
+
+        let synced_to_chain = node_status.latest_wallet_sync_timestamp.unwrap_or_default()
+            > esplora_chain_tip_timestamp
+            && node_status
+                .latest_onchain_wallet_sync_timestamp
+                .unwrap_or_default()
+                > esplora_chain_tip_timestamp;
+
+        Ok(GetNodeInfoResponse {
+            pub_key: self.node.node_id().serialize().to_vec(),
+            // TODO: This is a placeholder. We need to get the actual alias from the LDK node.
+            alias: format!("LDK Fedimint Gateway Node {}", self.node.node_id()),
+            network: match self.node.config().network {
+                Network::Bitcoin => "main",
+                Network::Testnet => "test",
+                Network::Signet => "signet",
+                Network::Regtest => "regtest",
+                _ => panic!("Unsupported network"),
+            }
+            .to_string(),
+            block_height,
+            synced_to_chain,
+        })
+    }
+
+    async fn routehints(
+        &self,
+        _num_route_hints: usize,
+    ) -> Result<GetRouteHintsResponse, LightningRpcError> {
+        // TODO: Return real route hints. Not strictly necessary but would be nice to
+        // have.
+        Ok(GetRouteHintsResponse {
+            route_hints: vec![],
+        })
+    }
+
+    async fn pay(
+        &self,
+        invoice: PayInvoiceRequest,
+    ) -> Result<PayInvoiceResponse, LightningRpcError> {
+        let parsed_invoice: Bolt11Invoice =
+            Bolt11Invoice::from_str(&invoice.invoice).map_err(|e| {
+                LightningRpcError::FailedPayment {
+                    failure_reason: format!("Failed to parse invoice: {e:?}"),
+                }
+            })?;
+
+        let payment_id = match self.node.bolt11_payment().send(&parsed_invoice) {
+            Ok(payment_id) => payment_id,
+            Err(e) => {
+                return Err(LightningRpcError::FailedPayment {
+                    failure_reason: format!("LDK payment failed to initialize: {e:?}"),
+                });
+            }
+        };
+
+        // TODO: Find a way to avoid looping/polling to know when a payment is
+        // completed. `ldk-node` provides `PaymentSuccessful` and `PaymentFailed`
+        // events, but interacting with the node event queue here isn't
+        // straightforward.
```

Comment:
If you want to implement this, you could potentially use a channel to communicate from the event queue thread to here. Can be done in a follow up